src/test/java/mate/academy/controller/ShoppingCartControllerIntegrationTest.java

```diff
@@ -0,0 +1,205 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import mate.academy.dto.cartitem.CartItemRequestDto;
+import mate.academy.dto.cartitem.CartItemResponseDto;
+import mate.academy.dto.cartitem.UpdateRequestDto;
+import mate.academy.dto.shoppingcart.ShoppingCartDto;
+import mate.academy.model.Book;
+import mate.academy.model.CartItem;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.User;
+import mate.academy.util.TestDataUtil;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class ShoppingCartControllerIntegrationTest {
+    private static final long VALID_ID = 1L;
+    private static final long INVALID_ID = -1L;
+    private static final String USER = "bob@in.ua";
+    private static User user;
+    private static Book book;
+    private static CartItem cartItem;
+    private static CartItemResponseDto cartItemResponseDto;
+    private static CartItemRequestDto cartItemRequestDto;
+    private static CartItemRequestDto invalidRequestDto;
+    private static UpdateRequestDto updateRequestDto;
+    private static ShoppingCart shoppingCart;
+    private static ShoppingCartDto shoppingCartDto;
+    private static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setUp(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+
+        user = TestDataUtil.getDefaultUser();
+        book = TestDataUtil.getDefaultBook();
+        cartItem = TestDataUtil.getDefaultCartItem();
+        cartItemRequestDto = TestDataUtil.getDefaultCartItemRequestDto();
+        cartItemResponseDto = TestDataUtil.getDefaultCartItemResponseDto();
+        invalidRequestDto = TestDataUtil.getBookInvalidRequestDto();
+        updateRequestDto = TestDataUtil.getDefaultUpdateRequestDto();
+        shoppingCart = TestDataUtil.getDefaultShoppingCart();
+        shoppingCartDto = TestDataUtil.getDefaultShoppingCartDto();
```

Comment:
I'd also made static imports for these methods to omit name of class

Reply:
Many thanks, I'll do it!